src/js/components/CheckBoxGroup/CheckBoxGroup.js

```diff
@@ -0,0 +1,99 @@
+import React, { forwardRef, useContext, useMemo } from 'react';
+
+import { Box } from '../Box';
+import { CheckBox } from '../CheckBox';
+import { FormContext } from '../Form/FormContext';
+
+export const CheckBoxGroup = forwardRef(
+  (
+    {
+      checked: checkedProp,
+      disabled: disabledProp,
+      gap = 'small', // consistent with RadioButtonGroup default
+      labelKey,
+      valueKey,
+      onChange,
+      options: optionsProp,
+      name,
+      ...rest
+    },
+    ref,
+  ) => {
+    const formContext = useContext(FormContext);
+
+    // In case option is a string, normalize it to be an object
+    const options = useMemo(
+      () =>
+        optionsProp.map(option => {
+          return typeof option === 'string'
+            ? {
+                disabled: disabledProp,
+                id: option,
+                label: option,
+              }
+            : option;
+        }),
+      [optionsProp, disabledProp],
+    );
+
+    // 'checked' is an array of checked valueKeys
+    const [checked, setChecked] = formContext.useFormContext(
+      name,
+      checkedProp,
+      '',
+    );
+
+    // Logic is necessary to maintain a proper data structure for Form logic
+    const onCheckBoxChange = (event, optionChecked) => {
+      const nextChecked = checked || [];
+      const optionIndex = nextChecked.indexOf(optionChecked);
+
+      if (optionIndex < 0) nextChecked.push(optionChecked);
+      else nextChecked.splice(optionIndex, 1);
+
+      setChecked(nextChecked);
+      if (onChange) onChange(event);
```

Comment:
We want to pass the same kind of thing we'll receive in the `checked` property as part of the event, so the caller can save this in his state and pass it back as `checked`.  Take a look at this from Select.js:
```
if (onChange) {
  event.persist();
  const adjustedEvent = event;
  adjustedEvent.target = inputRef.current;
  adjustedEvent.value = nextValue;
  adjustedEvent.option = option;
  adjustedEvent.selected = nextSelected;
  onChange(adjustedEvent);
}
```

Reply:
Done.